enterprise/server/backends/userdb/userdb.go

```diff
@@ -775,6 +775,27 @@ func (d *UserDB) InsertUser(ctx context.Context, u *tables.User) error {
 }
 
 func (d *UserDB) GetUserByID(ctx context.Context, id string) (*tables.User, error) {
+	var user *tables.User
+	err := d.h.Transaction(ctx, func(tx interfaces.DB) error {
+		u, err := d.getUser(ctx, tx, id)
+		if err != nil {
+			return err
+		}
+		user = u
+		return err
+	})
+	if err != nil {
+		return nil, err
+	}
+	for _, g := range user.Groups {
+		if err := perms.AuthorizeGroupAccess(ctx, d.env, g.Group.GroupID); err == nil {
+			return user, nil
+		}
+	}
```

Comment:
I'm a bit confused by what's going on here - could you add a comment?

(To me this new block seems the same as checking `len(user.Groups) > 0` - do we now need a stronger guarantee that users have to be part of at least one group?)

Reply:
There's two users involved, the authenticated user and the user being looked up. This is verifying that the authenticated user is in any of the same groups as the user being looked up.

Reply:
I see. Would it make sense to also check that the authenticated user is an admin of the group that they have in common with the user being looked up? Today, only admins have userdb-level access for other users.

Reply:
Done. This also required promoting SCIM API keys to Admin role (see the change in claims.go). 
I had that change in another PR because it's necessary for other group operations, but I moved it into this one since otherwise some tests fail.